test/unit/lodash.spec.js

```diff
@@ -0,0 +1,709 @@
+            expect(elem2 instanceof Object).to.eql(true);
+        });
+
+        it('should check if first object has right properties', function(){
+            expect(_.assign(params[0])).to.have.ownProperty('number');
+            expect(params[0].number).to.equal(5);
+            expect(_.assign(params[0])).to.have.ownProperty('square', 25);
+            expect(_.assign(params[0])).to.have.ownProperty('ping');
+            expect(_.assign(params[0])).to.have.ownProperty('pong');
+        });
+        
+        it('should check if obj function (ping) return right value', function(){
+            expect(params[0].ping()).to.eql('Number: ' + params[0].number);
+        });
+        
+        it('should check if obj function (pong) return right value', function(){
+            expect(params[0].pong()).to.eql('Number: ' + params[0].square);
+        });
+        
+        it('should check if Cube have right properties', function(){
+            expect(_.assign(params[1])).to.have.ownProperty('number', 6);
+        });
+        
+        it('should check if Cube have right prototype properties', function(){
+            expect(_.assign(params[1])).to.have.property('square', 36);
+            expect(_.assign(params[1])).not.to.have.ownProperty('square', 36);
+        });
+
+        it('should return new object with properties both objects', function(){
+            expect(_.assign.apply(_, params)).to.have.ownProperty('number', 6);
+            expect(_.assign.apply(_, params)).to.have.ownProperty('square', 25);
+            expect(_.assign.apply(_, params)).to.have.ownProperty('string', "Trust me! I am a engineer!")
+            expect(_.assign.apply(_, params).ping()).to.eql('Number: 6');
+            expect(_.assign.apply(_, params).pong()).to.eql('Number: 25');
+        });
+
+    });
+
+    describe('assignIn', function () {
+
+        var params;
+        beforeEach(function(){
+            params = datasets.assignIn();
+        });
+
+        it('should check types ', function(){
+            var elem1 = params[0];
+            var elem2 = params[1];
+
+            expect(typeof elem1).to.eql('object');
+            expect(elem2 instanceof Object).to.eql(true);
+        });
+
+        it('should check properties obj1', function(){
+            expect(_.assign(params[0])).to.have.ownProperty('number', 5);
+            expect(_.assign(params[0])).to.have.ownProperty('square');
+            expect(params[0].square()).to.eql('Square: ' + Math.pow(Math, this.number));
+        });
+
+        it('should check properties obj2', function(){
+            expect(_.assign(params[1])).to.have.ownProperty('number', 4);
+            expect(_.assign(params[1])).to.have.ownProperty('cube', 64);
+            expect(_.assign(params[1])).to.have.property('square');
+            expect(_.assign(params[1])).not.to.have.ownProperty('square');
+            expect(params[1].square()).to.eql('Square: ' + Math.pow(Math, this.number));
+        });
```

Comment:
If I were to say something, if you are using results from the same invocation in multiple test cases you should set it to temporary var(or ES2015 const, let but for now don't do that since our system isn't compatible yet).

I.e.
var assigmentResults = _.assign(params[1])

expect(assigmentResults).not.to.have.ownProperty('square');